File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.ArrayDeque;
import java.util.Deque;

public class Solution {
        public int[] solution(int []arr) {

            Deque<Integer> stacks = new ArrayDeque<>();
            int i =0;
            stacks.add(arr[i++]);
            for (;i<arr.length;i++) {
                 if (stacks.peekLast() == arr[i]){
                    continue;
                }
                stacks.add(arr[i]);
            }

            return stacks.stream().mapToInt(Integer::intValue).toArray();
        }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> sk;
    sk.push_back(arr[0]);

    for(int a : arr){
        if(sk.back()!=a){
            sk.push_back(a);
        }
    }
    return sk;
}
